### PR TITLE
Fix backup of external tables with --single-data-file

### PR DESCRIPTION
**Pull request by chrishajas**
During the gpbackup --single-data-file refactor to use an agent
approach, the oid list created for the master included external tables,
while the oid list on the segments did not. This caused the backup to
fail as the pipe names did not match (the segment created a pipe with
the name of a regular table, whereas the master executed a COPY TO
command to a pipe with an external table). This only occurred when
--single-data-file was used.

Authored-by: Chris Hajas <chajas@pivotal.io>

### CODE REVIEW

File: end_to_end/end_to_end_suite_test.go [hunk 1]
```diff
@@ -456,22 +472,6 @@ var _ = Describe("backup end to end integration tests", func() {
 
 			os.RemoveAll(backupdir)
 		})
-		It("runs gpbackup and gprestore on database with all objects", func() {
```
Comment (khuddlefish) [inline end_to_end/end_to_end_suite_test.go]: Do we not have one of these that runs without single-data-file now?

Comment (chrishajas) [reply end_to_end/end_to_end_suite_test.go]: Correct. We still have tests that run without `--single-data-file`, but the one containing all objects is now in `--single-data-file`. The metadata portion is identical, and really that's the bulk of what's being covered in this test. However, the single-data-file has some corner cases that will be covered better here. The other option is to do this test in both places, but it's ~20 seconds.

File: end_to_end/end_to_end_suite_test.go [hunk 2]
```diff
@@ -351,6 +351,22 @@ var _ = Describe("backup end to end integration tests", func() {
...
+				testutils.ExecuteSQLFile(backupConn, "gpdb4_objects.sql")
+				if backupConn.Version.AtLeast("5") {
+					testutils.ExecuteSQLFile(backupConn, "gpdb5_objects.sql")
+				}
+				if backupConn.Version.AtLeast("6") {
+					testutils.ExecuteSQLFile(backupConn, "gpdb6_objects.sql")
+				}
+				timestamp := gpbackup(gpbackupPath, "--leaf-partition-data", "--single-data-file")
+				gprestore(gprestorePath, timestamp, "--redirect-db", "restoredb")
+
```
Comment (soumyadeep2007) [inline end_to_end/end_to_end_suite_test.go]: There is no assert here?

Comment (khuddlefish) [reply end_to_end/end_to_end_suite_test.go]: The gpbackup and gprestore commands will fail the test if there is some problem. It is an implied assert that they run successfully.
